File: 326 Power of Three.c
bool isPowerOfThree(int n) {
    if(n <= 0) return false;
    long i = 1;
    while (i <= n )
    {
        if (i == n) 
            return true;
        else
        {
            i *= 3;
        }
    }
    return false;
}

File: 326 Power of Three.java
class Solution {
    public boolean isPowerOfThree(int n) {
        if (n <= 0)
            return false;
        while (n % 3 == 0)
            n /= 3;
        return n == 1;
    }
}

// 	Accepted	11 ms	38.9 MB	java
